src/main/java/de/tum/in/www1/artemis/service/WebsocketMessagingService.java

```diff
@@ -109,60 +113,109 @@ public CompletableFuture<Void> sendMessageToUser(String user, String topic, Obje
         }
     }
 
+    /**
+     * Broadcast a new result to the client.
+     * Waits until all notifications are sent and the result properties are restored.
+     * This allows the caller to reuse the passed result object again after calling.
+     *
+     * @param participation used to find the receivers of the notification
+     * @param result        the result object to publish
+     */
+    public void awaitBroadcastNewResult(Participation participation, Result result) {
+        try {
+            // Wait until all notifications got send and the objects were reconnected.
+            broadcastNewResult(participation, result).get();
+        }
+        catch (InterruptedException | ExecutionException e) {
+            log.warn("timed out while sending a result notification", e);
```

Comment:
Do the exceptions always mean a timeout? I feel like `ExecutionException` could be used for other problems, just guessing from the name 🤔

Reply:
true, `ExecutionException – if this future completed exceptionally` according to the Javadoc.